velox/functions/sparksql/Arithmetic.h

```diff
@@ -352,6 +352,46 @@ struct ToHexBigintFunction {
 };
 
 namespace detail {
+FOLLY_ALWAYS_INLINE bool
+isNull(double value, double min, double max, int64_t numBucket) {
+  return numBucket <= 0 || numBucket == std::numeric_limits<int64_t>::max() ||
+      std::isnan(value) || min == max || !std::isfinite(min) ||
+      !std::isfinite(max);
+}
+
+FOLLY_ALWAYS_INLINE int64_t computeBucketNumberNotNull(
+    double value,
+    double min,
+    double max,
+    int64_t numBucket) {
+  double lower = std::min(min, max);
+  double upper = std::max(min, max);
+
+  if (min < max) {
+    if (value < lower) {
+      return 0;
+    }
+
+    if (value >= upper) {
+      return numBucket + 1;
+    }
+    return static_cast<int64_t>(
+               (numBucket * (value - lower) / (upper - lower))) +
```

Comment:
A truncating or rounding up conversion from double to int64_t, which do we need to algin with Spark?

Reply:
hey @rui-mo the spark firstly covert `numBucket` to double to calculate the result and then convert the result to long, above c++ implement also do the same thing (interger will be implicitly converted to double) to calculate the result and then convert to int64, so what else should i change ?

Reply:
I see. Thanks for the explain.